### PR TITLE
Optionally output cytoscape json files

### PR DESCRIPTION
**Pull request by babayagaofficial**
Allows the user to optionally output the html visualisations, json files to be manually loaded into cytoscape, or both. The default is just the html visualisations. For the json files, also included in templates a style file in xml format to be used in cytoscape that is similar to the format used in the html visualisations (although stars are replaced with diamonds as cytoscape doesn't seem to allow for stars as node shape).

### CODE REVIEW

File: plasnet/output_producer.py [hunk 1]
```diff
@@ -11,40 +12,61 @@
...
+                json.dump(json_dict, file)
 
     @staticmethod
-    def produce_communities_visualisation(communities: Communities, outdir: Path) -> None:
-        OutputProducer._write_html_for_all_subgraphs(communities, outdir)
-        OutputProducer._produce_index_file(outdir, communities, "Communities")
+    def produce_communities_visualisation(
+        communities: Communities, outdir: Path, output_type: str
+    ) -> None:
+        OutputProducer._write_html_for_all_subgraphs(communities, outdir, output_type)
```
Comment (leoisl) [inline plasnet/output_producer.py:36]: Should this be called even if `output_type == json`?

File: plasnet/output_producer.py [hunk 2]
```diff
@@ -11,40 +12,61 @@
...
+            OutputProducer._produce_index_file(outdir, communities, "Communities")
 
     @staticmethod
     def produce_subcommunities_visualisation(
-        subcommunities: ListOfGraphs[HubGraph], outdir: Path
+        subcommunities: ListOfGraphs[HubGraph], outdir: Path, output_type: str
     ) -> None:
-        OutputProducer._write_html_for_all_subgraphs(subcommunities, outdir)
-        OutputProducer._produce_index_file(outdir, subcommunities, "subcommunity")
+        OutputProducer._write_html_for_all_subgraphs(subcommunities, outdir, output_type)
```
Comment (leoisl) [inline plasnet/output_producer.py:44]: Should this be called even if `output_type == json`?

**Review by leoisl — CHANGES_REQUESTED** [review]
Hey @babayagaofficial , thanks for this PR, I really appreciate it! Just two minor questions!

File: plasnet/output_producer.py (hunk 1 above)
Comment (babayagaofficial) [reply plasnet/output_producer.py:36]: I've actually hijacked this function to write both json and html as needed. Might be worth renaming the function to something like `write_vis_for_all_subgraphs` for clarity?

**Comment by leoisl** [comment]
LGTM, thanks a lot @babayagaofficial !

**Pull request merged**